react-tic-tac-toe/src/app/components/App/game/index.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+
+import Board from '../board';
+import { calculateWinner } from '../helpers';
+
+import styles from './styles.scss';
+
+export default class Game extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      history: [
+        {
+          squares: Array(9).fill(null)
+        }
+      ],
+      stepNumber: 0,
+      xIsNext: true
+    };
+  }
+
+  handleClick(i) {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[this.state.stepNumber];
+    const squares = current.squares.slice();
+    if (calculateWinner(squares) || squares[i]) {
+      return;
+    }
+    squares[i] = this.state.xIsNext ? 'X' : 'O';
+    this.setState({
+      history: history.concat([
+        {
+          squares
+        }
+      ]),
+      stepNumber: history.length,
+      xIsNext: !this.state.xIsNext
```

Comment:
When your next state depends on the previous one you should use the function setState sintax:
```js
this.setState((prevState) => ({
  xIsNext: !prevState.xIsNext
});
```

Read more here https://reactjs.org/docs/react-component.html#setstate